### PR TITLE
[helper-plugin] create unit test for useFocusWhenNavigate

### PR DESCRIPTION
**Pull request by yurimutti**
### What does it do?

Add unit test for useFocusWhenNavigate hook with typescript

### Why is it needed?

useFocusWhenNavigate needed a unit test for more quality code.

### How to test it?

yarn test:front 

### Related issue(s)/PR(s)

[related issue](https://github.com/strapi/strapi/issues/17690)

### CODE REVIEW

File: packages/core/helper-plugin/src/hooks/tests/useFocusWhenNavigate.test.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import { useFocusWhenNavigate } from '../useFocusWhenNavigate';
+
+const consoleWarnMock = jest.spyOn(console, 'warn').mockImplementation();
+
+const setup = () => {
+  return render(
+    <ThemeProvider theme={lightTheme}>
+      <IntlProvider locale="en" messages={{}} defaultLocale="en">
+        <Main />
```
Comment (markkaylor) [inline packages/core/helper-plugin/src/hooks/tests/useFocusWhenNavigate.test.tsx:15]: I just have an error here for the labelledBy prop, do you mind adding it?

**Review by markkaylor — COMMENTED** [review]
The test looks good to me. Just one small change to get rid of an error if you don't mind. Thanks!

File: packages/core/helper-plugin/src/hooks/tests/useFocusWhenNavigate.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { IntlProvider } from 'react-intl';
+
+import { useFocusWhenNavigate } from '../useFocusWhenNavigate';
+
+const consoleWarnMock = jest.spyOn(console, 'warn').mockImplementation();
+
+const setup = () => {
+  return render(
+    <ThemeProvider theme={lightTheme}>
+      <IntlProvider locale="en" messages={{}} defaultLocale="en">
```
Comment (markkaylor) [inline packages/core/helper-plugin/src/hooks/tests/useFocusWhenNavigate.test.tsx:14]: I believe you can remove these, similar to the suggestion on:

https://github.com/strapi/strapi/pull/17992/files#r1321089293

File: packages/core/helper-plugin/src/hooks/tests/useFocusWhenNavigate.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable check-file/filename-naming-convention */
+
+import { lightTheme, ThemeProvider, Main } from '@strapi/design-system';
+import { render, renderHook, screen } from '@testing-library/react';
```
Comment (markkaylor) [inline packages/core/helper-plugin/src/hooks/tests/useFocusWhenNavigate.test.tsx:4]: I think you can do the same suggestion as: https://github.com/strapi/strapi/pull/17992/files#r1321089847

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**